### PR TITLE
adding support to filter /api/v1/rules response

### PR DESCRIPTION
**Pull request by victoramsantos**
### Is your feature request related to a problem? Please describe

By the [Prometheus doc](https://prometheus.io/docs/prometheus/latest/querying/api/#rules) we have the possibility to filter the rules by alert or record rules (using the `type` param). On the [vm-alert doc](https://docs.victoriametrics.com/vmalert/#web) it's just saying that this api list of all loaded groups and rules . When I run with the type it returns the same info for both param value.

### Describe the solution you'd like

I would like that this API have the same behavior as in Prometheus, if I pass the parameter `?type=alert` it would return only the alerts, if I pass `?type=record` only the record rules and if it's without `type` both data.

### Describe alternatives you've considered

_No response_

### Additional information

- I know it's possible to iterate over the response and filter each rule however I trying to make a cardinality analyzer that works seamless between Prometheus and VM
- I didn't change the `vmalert/api/v1/rules` endpoint, since I don't know if it's desired to it be equal to `api/v1/rules`. Feedbacks here please 😊 

Related feature-request: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5743

### CODE REVIEW

**Comment by Haleygo** [comment]
Hey @victoramsantos , thanks for the PR!
>I didn't change the vmalert/api/v1/rules endpoint, since I don't know if it's desired to it be equal to api/v1/rules.

Yes, they should behave the same.
And since we're adding this, I think below apis could share the same extra filter parameter:
1. `/vmalert/groups`
2. `/rules`
3. `/vmalert/api/v1/rules`, `/api/v1/rules`

And I think we can add this filter parameter in
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/de9a9546c2ee6449d291aaf863a68843b67acebc/app/vmalert/web.go#L210

**Comment by victoramsantos** [comment]
Hi @Haleygo job done!
Please any concern or feedback are very welcome 😊

File: app/vmalert/web.go [hunk 1]
```diff
@@ -87,7 +87,14 @@ func (rh *requestHandler) handler(w http.ResponseWriter, r *http.Request) bool {
 		WriteRuleDetails(w, r, rule)
 		return true
 	case "/vmalert/groups":
-		WriteListGroups(w, r, rh.groups())
+		var data []apiGroup
+		ruleType := r.URL.Query().Get("type")
+		if ruleType == "alert" || ruleType == "record" {
```
Comment (Haleygo) [inline app/vmalert/web.go:92]: I don't think we need to have different function for null `ruleType` and non-null `rulesType` in routes, we can pass the `ruleType` to func `rh.groups()` to filter rules, like
```
func (rh *requestHandler) groups(ruleType string) []apiGroup {
	rh.m.groupsMu.RLock()
	defer rh.m.groupsMu.RUnlock()

	groups := make([]apiGroup, 0)
	for _, g := range rh.m.groups {
		g = g.DeepCopy()
		var matchedRules []rule.Rule
		if ruleType != "" {
			for _, r := range g.Rules {
				if _, ok := r.(*rule.AlertingRule); ok && ruleType == "alert" {
					matchedRules = append(matchedRules, r)
				}
				if _, ok := r.(*rule.RecordingRule); ok && ruleType == "record" {
					matchedRules = append(matchedRules, r)
				}
			}
			if len(matchedRules) == 0 {
				continue
			}
			g.Rules = matchedRules
		}
		groups = append(groups, groupToAPI(g))
	}
```
and `rh.listGroups()` will call `rh.groups()` as well, so no need to add below new function.
@victoramsantos wdyt?

**Comment by victoramsantos** [comment]
Yes, I like it :)
I just changed the `if ruleType != "" ` because for cases where the type is wrong I think we should return ignore it and return everything. Do you agree? It's the same behavior I found for Prometheus' API.

**Comment by Haleygo** [comment]
Looks good to me!
Could you help to add a feature description to the [changelog](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/CHANGELOG.md)?

**Comment by victoramsantos** [comment]
@Haleygo I think I did it right 😅

**Review by Haleygo — APPROVED** [review]
LGTM!
Thanks!

**Comment by victoramsantos** [comment]
Nice! @Haleygo I need to do something else or just wait for the merge?

**Comment by hagen1778** [comment]
Please also see https://github.com/VictoriaMetrics/VictoriaMetrics/pull/5787

**Comment by valyala** [comment]
FYI, this pull request has been included in [vmalert v1.98.0](https://docs.victoriametrics.com/CHANGELOG.html).

**Pull request merged**